### ISSUE TITLE
Using CrossEntropyLoss in the output layer

### ISSUE THREAD

**Issue by nikrao** [dmlc/mxnet#5900]
I am trying to use the CrossEntropyLoss() layer as an output, and get an error saying it needs 2 inputs. (data and label). But when I instead use SoftmaxOutput, I only need to provide the data, and the label is provided via the training iterator. Is there a way I can use the cross entropy loss with only providing data?

## Minimum reproducible example

depth = 3

	data = mx.sym.Variable('data')
	# layer 1 is as defined
	net = mx.sym.FullyConnected(data=data,name='fc1',num_hidden=64)
	net = mx.sym.Activation(data=net,name='ac1',act_type="relu")
	net = mx.sym.Dropout(data=net,name='dl1',p=0.1)
	# remaining hidden layers
	for layer in xrange(2,depth+1):
		net = mx.sym.FullyConnected(data=net,name='fc'+str(layer),num_hidden=64)
		net = mx.sym.Activation(data=net,name='ac'+str(layer),act_type="relu")
		net = mx.sym.Dropout(data=net,name='dl'+str(layer),p=0.1)
	# output layer
	net = mx.sym.FullyConnected(data=net,name='fcout',num_hidden=10)
	net = mx.sym.SoftmaxOutput(data=net,name='softmax')
	#net = rt.CrossEntropyLoss(data=net)
	return net

uncommenting the second-last line above (and commenting the line above it) gives me the error

**Comment by jdhao** [comment]
@nikrao , there is no `CrossEntropyLoss` symbol in MXNet, `mx.sym.SoftmaxOutput()` implements the cross entropy loss, see [here](http://mxnet.io/api/python/symbol.html#mxnet.symbol.SoftmaxOutput).

**Comment by nikrao** [comment]
Hi Jeidong,

but this is only in the backward pass right? What if I only want it in a
feedforward network?

**Comment by jdhao** [comment]
@nikrao , what do you mean by "I only want it in a feedforward network"? I don't get it.

**Comment by nikrao** [comment]
sorry for the confusion. I mean the following:

I just have a simple feedforward NN, with, say 1 hidden layer. I guess I'm
confused with the page saying: "In the backward pass, the logit loss, also
called cross-entroy loss, is added" .

Does this mean that while training, this is actually the cross entropy loss
that we are trying to minimize in the objective function?

thanks

**Comment by jdhao** [comment]
@nikrao , yes, according to the documentation, in forward pass, this layers only computes the softmax output of previous layer (i.e., a normalized probability distribution). 
During training, the cross entropy loss is computed by combining this info and the ground truth label info.
See [here](http://mxnet.io/tutorials/python/mnist.html#) for a complete example on how to train a feedforward or CNN  on the minist dataset using `mx.sym.SoftmaxOutput`

**Comment by nikrao** [comment]
I checked this out, and this seems to be doing some funky version of softmax cross entropy, and not the traditional binary cross entropy. I am interested in the multilabel problem , so this will notwork. I have to define a new loss and add it in as referenced in other issues. 

I will try it an open a separate issue if that does not work
